src/Migration/Connection/Helper/ConnectionNameSanitizer.php

```diff
@@ -0,0 +1,23 @@
+<?php declare(strict_types=1);
+/*
+ * (c) shopware AG <info@shopware.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace SwagMigrationAssistant\Migration\Connection\Helper;
+
+use Shopware\Core\Framework\Log\Package;
+
+#[Package('fundamentals@after-sales')]
+class ConnectionNameSanitizer
+{
+    /**
+     * Sanitizes the connection name by removing any special characters, hyphens and spaces.
+     * Only alphanumeric characters are allowed.
+     */
+    public static function sanitize(string $connectionName): string
+    {
+        return \preg_replace('/[^a-zA-Z0-9]/', '', \str_replace([' ', '-'], '', $connectionName));
+    }
```

Comment:
LGTM 👍 , we could optionally mention why this constraint exists in the first place: we sometimes use the connection name in places where a unique technical name is needed and e.g. CustomFields have their own constraints on names, see
https://github.com/shopware/shopware/blob/df7ae4bf10431177298ed5bd8e32a6da0ba3c93f/src/Core/System/CustomField/CustomFieldService.php#L29